src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/BaseRepository.cs

```diff
@@ -0,0 +1,108 @@
+namespace Altinn.Platform.Storage.Repository
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Altinn.Platform.Storage.Configuration;
+    using Altinn.Platform.Storage.Extensions;
+    using Microsoft.Azure.Documents;
+    using Microsoft.Azure.Documents.Client;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Options;
+
+    /// <summary>
+    /// Base repository service for initializing db, collections etc.
+    /// </summary>
+    internal abstract class BaseRepository : IHostedService, IDisposable
```

Comment:
Nice extracting this into an abstract class! Saves us from all the duplicationa we had :)